TESTS/mbed_platform/error_handling/main.cpp

```diff
@@ -0,0 +1,373 @@
+        }
+            
+        //printf("\nError Status[%d] = 0x%08X Value = 0x%08X\n", i, (unsigned int)error_ctx.error_status, (unsigned int)error_ctx.error_value);
+        TEST_ASSERT_EQUAL_UINT((unsigned int)error_ctx.error_value, (unsigned int)error_ctx.error_status);
+    }
+}
+#endif
+
+static Semaphore    callback_sem;
+void MyErrorHook(const mbed_error_ctx *error_ctx)
+{
+    callback_sem.release();
+}
+
+/** Test error hook
+ */
+void test_error_hook()
+{
+    if( MBED_SUCCESS != mbed_set_error_hook(MyErrorHook)) {
+        TEST_FAIL();
+    }
+    
+    MBED_WARNING(MBED_ERROR_INVALID_ARGUMENT, "Test for error hook", 1234);
+    int32_t sem_status = callback_sem.wait(5000);
+    
+    TEST_ASSERT(sem_status > 0);
+}
+
+#ifdef MBED_TEST_SIM_BLOCKDEVICE
+
+// test configuration
+#ifndef MBED_TEST_FILESYSTEM
+#define MBED_TEST_FILESYSTEM LittleFileSystem
+#endif
+
+#ifndef MBED_TEST_FILESYSTEM_DECL
+#define MBED_TEST_FILESYSTEM_DECL MBED_TEST_FILESYSTEM fs("fs")
+#endif
+
+#ifndef MBED_TEST_BLOCK_COUNT
+#define MBED_TEST_BLOCK_COUNT 64
+#endif
+
+#ifndef MBED_TEST_SIM_BLOCKDEVICE_DECL
+#define MBED_TEST_SIM_BLOCKDEVICE_DECL MBED_TEST_SIM_BLOCKDEVICE fd(MBED_TEST_BLOCK_COUNT*512, 1, 1, 512)
+#endif
+
+// declarations
+#define STRINGIZE(x) STRINGIZE2(x)
+#define STRINGIZE2(x) #x
+#define INCLUDE(x) STRINGIZE(x.h)
+
+#include INCLUDE(MBED_TEST_FILESYSTEM)
+#include INCLUDE(MBED_TEST_SIM_BLOCKDEVICE)
+
+MBED_TEST_FILESYSTEM_DECL;
+MBED_TEST_SIM_BLOCKDEVICE_DECL;
+
+/** Test save error log
+ */
+void test_save_error_log()
+{
+    //Log some errors
+    MBED_WARNING(MBED_ERROR_TIME_OUT, "Timeout error", 1 );
+    MBED_WARNING(MBED_ERROR_ALREADY_IN_USE, "Already in use error", 2 );
+    MBED_WARNING(MBED_ERROR_UNSUPPORTED, "Not supported error", 3 );
+    MBED_WARNING(MBED_ERROR_ACCESS_DENIED, "Access denied error", 4 );
+    MBED_WARNING(MBED_ERROR_ITEM_NOT_FOUND, "Not found error", 5 );
+    
+    int error = 0;
+    
+    error = MBED_TEST_FILESYSTEM::format(&fd);
+    if(error < 0) {
+        printf("Failed formatting");
+        TEST_FAIL();
+    }
+    
+    error = fs.mount(&fd);
+    if(error < 0) {
+        printf("Failed mounting fs");
+        TEST_FAIL();
+    }
+    
+    if(MBED_SUCCESS != mbed_save_error_log("/fs/errors.log")) {
+        printf("Failed saving error log");
+        TEST_FAIL();
+    }
+    
+    FILE *error_file = fopen("/fs/errors.log", "r");
+    if(error_file == NULL) {
+        printf("Unable to find error log in fs");
+        TEST_FAIL();
+    }
+    
+    char buff[64] = {0};
+    while (!feof(error_file)){
+      int size = fread(&buff[0], 1, 15, error_file);
+      fwrite(&buff[0], 1, size, stdout);
+    }
+    printf("\r\n");
+    fclose(error_file);
+    
+    error = fs.unmount();
+    if(error < 0) {
+        printf("Failed unmounting fs");
+        TEST_FAIL();
+    }
+}
+
+#endif
+
+utest::v1::status_t test_setup(const size_t number_of_cases)
+{
+    GREENTEA_SETUP(300, "default_auto");
```

Comment:
Does the test take this long (300 sec) to finish?

Reply:
I noticed that it takes quite a long for some slow targets(I saw one of them took 80secs it could be some test delays as well) as I see. But I can cut that down to 2 mins or less. Ill do that.